workspaces/cli-server/src/diffs/initial-body-manager.ts

```diff
@@ -0,0 +1,56 @@
+import { EventEmitter } from 'events';
+import { runManagedScriptByName } from '@useoptic/cli-scripts';
+import { ChildProcess } from 'child_process';
+import {
+  getInitialBodiesOutputPaths,
+  IInitialBodiesProjectionEmitterConfig,
+} from '@useoptic/cli-shared/build/diffs/initial-bodies-worker';
+import { ILearnedBodies } from '@useoptic/cli-shared/build/diffs/initial-types';
+
+export interface IInitialBodyManagerConfig {
+  pathId: string;
+  method: string;
+  captureId: string;
+  captureBaseDirectory: string;
+}
+
+export class InitialBodyManager {
+  public readonly events: EventEmitter = new EventEmitter();
+  private child!: ChildProcess;
+
+  async run(config: IInitialBodyManagerConfig): Promise<ILearnedBodies> {
+    const outputPaths = getInitialBodiesOutputPaths(config);
+    const scriptConfig: IInitialBodiesProjectionEmitterConfig = {
+      captureId: config.captureId,
+      pathId: config.pathId,
+      method: config.method,
+      specFilePath: outputPaths.events,
+      captureBaseDirectory: config.captureBaseDirectory,
+    };
```

Comment:
We adapted the signature of this in `run` (renamed `start`) to no longer accept the config, but instead construct it itself. Ultimately, if the task requires any persistence to compute it's results, that should be an implementation detail to the task, not something an API route handler should have to worry about. See the comment in the route-handler for more info.

Here's how we've currently approached that for the diffing task, effectively just moving the logic in here: 

https://github.com/opticdev/optic/blob/19386cab002eae2af7db3d8fe3969ce1248ff005/workspaces/cli-server/src/diffs/on-demand.ts#L44-L64

The initial config is assigned at moment of construction: https://github.com/opticdev/optic/blob/19386cab002eae2af7db3d8fe3969ce1248ff005/workspaces/cli-server/src/diffs/on-demand.ts#L32-L35